### PR TITLE
Texture init observables

### PR DESCRIPTION
**Pull request by royibernthal**

### CODE REVIEW

**Comment by RaananW** [comment]
So, I suddenly wonder what's the use of the onAfter function. As almost everything happens async, it's practically the same. The existing onLoaded observer can be used instead.

**Comment by deltakosh** [comment]
I agree, Only onBefore is useful

File: src/Materials/Textures/babylon.texture.ts [hunk 1]
```diff
@@ -117,6 +117,8 @@
 
             scene = this.getScene();
 
+            scene.getEngine().onBeforeTextureInitObservable.notifyObservers(this);
```
Comment (deltakosh) [inline src/Materials/Textures/babylon.texture.ts]: we need to make sure that this.url is used later on by the constructor (and not just url)

**Comment by royibernthal** [comment]
True I didn't notice that. Check my new commit.

**Comment by deltakosh** [comment]
Sounds good!

**Comment by royibernthal** [comment]
Great :) Any idea when you're gonna build it for dist?

**Comment by deltakosh** [comment]
In the next hour :)

**Pull request merged**